packages/apollo-federation/src/service/__tests__/buildFederatedSchema.test.ts

```diff
@@ -10,40 +17,80 @@ const EMPTY_DOCUMENT = {
   definitions: [],
 };
 
+const createBuildFederatedSchemaTests = (
+  name: string,
+  schema: GraphQLSchema,
+  testSchema: (schema: GraphQLSchema) => Promise<void>,
+) => {
+  it(name, async () => {
+    await testSchema(schema);
+  });
+
+  it(`${name} (using "schema" argument)`, async () => {
+    await testSchema(buildFederatedSchema(schema));
+  });
+};
+
 describe('buildFederatedSchema', () => {
-  it(`should mark a type with a key field as an entity`, () => {
-    const schema = buildFederatedSchema(gql`
+  createBuildFederatedSchemaTests(
+    'should mark a type with a key field as an entity',
+    buildFederatedSchema(gql`
       type Product @key(fields: "upc") {
         upc: String!
         name: String
         price: Int
       }
-    `);
+    `),
+    async (schema: GraphQLSchema) => {
+      expect(schema.getType('Product')).toMatchInlineSnapshot(`
+type Product {
+  upc: String!
+  name: String
+  price: Int
+}
+`);
 
-    expect(schema.getType('Product')).toMatchInlineSnapshot(`
+      expect(schema.getType('_Entity')).toMatchInlineSnapshot(
+        `union _Entity = Product`,
+      );
+    },
+  );
+  createBuildFederatedSchemaTests(
```

Comment:
Is this test the same as above?

Reply:
@JacksonKearl yup, it is.  Fixed in latest push.